src/main/java/it/pagopa/pn/delivery/svc/NotificationReceiverService.java

```diff
@@ -122,21 +122,22 @@ public NewNotificationResponse receiveNotification(
 				.boxed()
 				.flatMap(recipientIndex -> {
 					NotificationRecipient notificationRecipient = internalNotification.getRecipients().get(recipientIndex);
-					return IntStream.range(0, notificationRecipient.getPayments().size())
-							.boxed()
-							.flatMap(paymentIndex -> {
-								NotificationPaymentInfo notificationPaymentInfo = notificationRecipient.getPayments().get(paymentIndex);
-								if (notificationPaymentInfo.getF24() != null) {
-									SaveF24Item saveF24Item = new SaveF24Item();
-									saveF24Item.setApplyCost(notificationPaymentInfo.getF24().isApplyCost());
-									saveF24Item.setSha256(notificationPaymentInfo.getF24().getMetadataAttachment().getDigests() != null ? notificationPaymentInfo.getF24().getMetadataAttachment().getDigests().getSha256() : null);
-									saveF24Item.setFileKey(notificationPaymentInfo.getF24().getMetadataAttachment().getRef() != null ? notificationPaymentInfo.getF24().getMetadataAttachment().getRef().getKey() : null);
-									saveF24Item.setPathTokens(List.of(Integer.toString(recipientIndex), Integer.toString(paymentIndex)));
-									return Stream.of(saveF24Item);
-								} else {
-									return Stream.empty();
-								}
-							});
+					if (CollectionUtils.isEmpty(notificationRecipient.getPayments())) {
```

Comment:
mi sembra sia invertito il controllo

Reply:
Dire di sistemare i test unitari... la causa di questo comportamento dovrebbe essere causato dall'uso improprio di questo metodo it.pagopa.pn.delivery.svc.NotificationReceiverTest#newNotificationWithoutPayments